src/ensembl/src/content/app/species/hooks/useSpeciesRouting.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useCallback } from 'react';
+import { useParams } from 'react-router-dom';
+import { replace } from 'connected-react-router';
+import { useSelector, useDispatch } from 'react-redux';
+
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import { getEnabledCommittedSpecies } from 'src/content/app/species-selector/state/speciesSelectorSelectors';
+import { getActiveGenomeId } from 'src/content/app/species/state/general/speciesGeneralSelectors';
+import { setActiveGenomeId } from 'src/content/app/species/state/general/speciesGeneralSlice';
+
+const useSpeciesRouting = () => {
```

Comment:
I am confused as to the purpose of this hook. It does have a purpose if we ever want the `/species` to be a valid route by itself; otherwise all the necessary information should just be retrievable from the url. Do we want the empty `/species` to be a valid url?

Reply:
We handle urls like `/genome-browser` and `/entity-viewer`.  Can we not do something similar here?

Reply:
We do so, because we have links to genome-browser and entiity-viewer that don't specify the species; but I don't expect us to have a link to /species without a species. The only way user can get to bare /species is if they manually delete the genome id from the url. At this point, it's kind of their fault; we can show a 404.

Unless I am missing something.

Reply:
Removed the hook and redirecting to species selector when there is no genome ID